### DISCUSSION TITLE
How stable is the interface of spdlog and what is the memory footprint of it

### DISCUSSION THREAD

**Discussion by rsniezek** [gabime/spdlog#2741] [Q&A] [answered]
Hi all,

first of all, thank you for the hard work and great library!

I'd like to integrate the lib into the project I work on in my job but our FOSS introduction procedure requires me to get some data. What I am missing is:
1. How stable is the interface of spdlog. How often the version released requires code changes in client SW to work. I assume that since we are still on version 1.XX, the answer is "very rarely" but I'd like to get confirmation.
2. Do you have release schedule for any upcoming changes that could be breaking the compatibility?
3. Is there any thought about the end of life for the library? I hope not :) 
4. What is the memory footprint of the library. I get the .so file size from my build but the question is about how much memory will the library allocate for it's needs. This can be an estimate, not concrete number. Or even list of things the lib will use memory for so that I could estimate the footprint for my usecase.

Thank you in advance for all the answers

**Comment by gabime** [comment, marked as answer]
1. Since we use semver, we **try** hard not to break the API without changing the major version number, but of course, nothing is foolproof. 
2. No
3. No, though rate of development and support vary according to the mainteiners  availability(like in any foss i guess).
4. It really depends on the usecae. I suggest measure with your specific usecase and platform.

**Reply by rsniezek** [reply]
Thank you. That settles it
